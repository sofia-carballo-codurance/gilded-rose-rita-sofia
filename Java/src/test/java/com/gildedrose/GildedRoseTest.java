package com.gildedrose;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class GildedRoseTest {

    @Test
    void should_update_quality() {
        Item[] items = new Item[] { new Item("foo", 0, 0) };
        GildedRose app = new GildedRose(items);
        app.updateQuality();

        Approvals.verify(app.items[0].toString());
    }
}
