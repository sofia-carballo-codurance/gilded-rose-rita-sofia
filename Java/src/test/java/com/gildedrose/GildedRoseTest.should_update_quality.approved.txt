[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 30] => Aged Brie, -1, 32 
[Aged Brie, 0, 50] => Aged Brie, -1, 50 
[Aged Brie, 0, 60] => Aged Brie, -1, 60 
[Aged Brie, 1, 0] => Aged Brie, 0, 1 
[Aged Brie, 1, 30] => Aged Brie, 0, 31 
[Aged Brie, 1, 50] => Aged Brie, 0, 50 
[Aged Brie, 1, 60] => Aged Brie, 0, 60 
[Aged Brie, 6, 0] => Aged Brie, 5, 1 
[Aged Brie, 6, 30] => Aged Brie, 5, 31 
[Aged Brie, 6, 50] => Aged Brie, 5, 50 
[Aged Brie, 6, 60] => Aged Brie, 5, 60 
[Aged Brie, 11, 0] => Aged Brie, 10, 1 
[Aged Brie, 11, 30] => Aged Brie, 10, 31 
[Aged Brie, 11, 50] => Aged Brie, 10, 50 
[Aged Brie, 11, 60] => Aged Brie, 10, 60 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 30] => Sulfuras, Hand of Ragnaros, 0, 30 
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras, Hand of Ragnaros, 0, 50 
[Sulfuras, Hand of Ragnaros, 0, 60] => Sulfuras, Hand of Ragnaros, 0, 60 
[Sulfuras, Hand of Ragnaros, 1, 0] => Sulfuras, Hand of Ragnaros, 1, 0 
[Sulfuras, Hand of Ragnaros, 1, 30] => Sulfuras, Hand of Ragnaros, 1, 30 
[Sulfuras, Hand of Ragnaros, 1, 50] => Sulfuras, Hand of Ragnaros, 1, 50 
[Sulfuras, Hand of Ragnaros, 1, 60] => Sulfuras, Hand of Ragnaros, 1, 60 
[Sulfuras, Hand of Ragnaros, 6, 0] => Sulfuras, Hand of Ragnaros, 6, 0 
[Sulfuras, Hand of Ragnaros, 6, 30] => Sulfuras, Hand of Ragnaros, 6, 30 
[Sulfuras, Hand of Ragnaros, 6, 50] => Sulfuras, Hand of Ragnaros, 6, 50 
[Sulfuras, Hand of Ragnaros, 6, 60] => Sulfuras, Hand of Ragnaros, 6, 60 
[Sulfuras, Hand of Ragnaros, 11, 0] => Sulfuras, Hand of Ragnaros, 11, 0 
[Sulfuras, Hand of Ragnaros, 11, 30] => Sulfuras, Hand of Ragnaros, 11, 30 
[Sulfuras, Hand of Ragnaros, 11, 50] => Sulfuras, Hand of Ragnaros, 11, 50 
[Sulfuras, Hand of Ragnaros, 11, 60] => Sulfuras, Hand of Ragnaros, 11, 60 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 30] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 60] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 1, 0] => Backstage passes to a TAFKAL80ETC concert, 0, 3 
[Backstage passes to a TAFKAL80ETC concert, 1, 30] => Backstage passes to a TAFKAL80ETC concert, 0, 33 
[Backstage passes to a TAFKAL80ETC concert, 1, 50] => Backstage passes to a TAFKAL80ETC concert, 0, 50 
[Backstage passes to a TAFKAL80ETC concert, 1, 60] => Backstage passes to a TAFKAL80ETC concert, 0, 60 
[Backstage passes to a TAFKAL80ETC concert, 6, 0] => Backstage passes to a TAFKAL80ETC concert, 5, 2 
[Backstage passes to a TAFKAL80ETC concert, 6, 30] => Backstage passes to a TAFKAL80ETC concert, 5, 32 
[Backstage passes to a TAFKAL80ETC concert, 6, 50] => Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Backstage passes to a TAFKAL80ETC concert, 6, 60] => Backstage passes to a TAFKAL80ETC concert, 5, 60 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => Backstage passes to a TAFKAL80ETC concert, 10, 1 
[Backstage passes to a TAFKAL80ETC concert, 11, 30] => Backstage passes to a TAFKAL80ETC concert, 10, 31 
[Backstage passes to a TAFKAL80ETC concert, 11, 50] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 60] => Backstage passes to a TAFKAL80ETC concert, 10, 60 
